arelle/plugin/validate/ESEF/__init__.py

```diff
@@ -131,14 +131,16 @@ def modelDocumentPullLoader(
                     if documentType in {ModelDocument.Type.TESTCASESINDEX, ModelDocument.Type.TESTCASE}:
                         return None  # allow zipped test case to load normally
 
-                if disclosureSystemYear >= 2023 and not modelXbrl.fileSource.isZip:
+                isZipFormat = modelXbrl.fileSource.isZip
+                hasZipFileExtension = modelXbrl.fileSource.type.lower() == ".zip"
+                if disclosureSystemYear >= 2023 and not (isZipFormat and hasZipFileExtension):
```

Comment:
Do I understand that `.isZip` previously only returned true if the suffix was `.zip`, but we're changing it to return true for any zip archive, so we're adjusting this code to maintain the same semantics as before?  I wonder if that change in behavior would have any impact on plugins(?) and whether a different `.isZipArchive` would be preferable to changing the meaning of `.isZip`.

To be sure, I suppose some subsequent PR would change the logic here to allow `.xbri` here too, right?

Reply:
> Do I understand that .isZip previously only returned true if the suffix was .zip, but we're changing it to return true for any zip archive, so we're adjusting this code to maintain the same semantics as before?

Correct.

> I wonder if that change in behavior would have any impact on plugins(?) and whether a different .isZipArchive would be preferable to changing the meaning of .isZip.

I conducted a search across Arelle and all known plugins, both within the Arelle repository and externally. The direct use of `isZip` is quite limited. Most other uses are actually checking if the file source is in a zip archive format. The more common usage is the `isArchive` property, which includes `isZip`, or `isTarGz`, etc. These uses focus on the file format rather than the file extension.

> To be sure, I suppose some subsequent PR would change the logic here to allow .xbri here too, right?

Yes, this change, along with others, will be necessary to complete support for ESEF 2024.